web-api/src/persistence/dynamo/helpers/purgeDynamoKeys.ts

```diff
@@ -1,11 +1,11 @@
 const dynamoKeys = ['pk', 'sk', 'gsi1pk', 'gsi2pk', 'ttl'];
-export function purgeDynamoKeys<T>(obj: T): T {
+export function purgeDynamoKeys<T, S>(obj: T): S {
   for (const property in obj) {
     if (dynamoKeys.includes(property)) {
       delete obj[property];
     } else if (typeof obj[property] === 'object') {
       obj[property] = purgeDynamoKeys(obj[property]);
     }
   }
-  return obj;
+  return obj as unknown as S;
```

Comment:
We aren't returning type T--we are returning a new, unknown type. This is hardly used anymore; I allowed specifying the return type S for ease of use.